### DISCUSSION TITLE
About Twitch Chat Commands

### DISCUSSION THREAD

**Discussion by jetfuel** [TwitchIO/TwitchIO#421] [Questions] [answered]
Language: Python

Hi, first time here. I spent some time studying Twitch API and tmi.js. I am able to build a simple bot (`commands.Bot`) that can read/write message using Twitchio. 

But when I tried to utilize the "Twitch Chat Commands" within the bot (like creating announcement with /announce or clear chat with /clear), I was unable to achieve that. 

I tried to send out plan Twitch Chat Commands, for example, `ctx.send('/clear')` just to see if I can execute Twitch Chat Commands like that. Needless to say, the call didn't work. Twitch said that they will stop supporting IRC command. Not sure if that's the reason. [link](https://discuss.dev.twitch.tv/t/deprecation-of-chat-commands-through-irc/40486)

From the reference doc, I found there is a class `twitchio.PartialUser` that has a function 'chat_announcement', but I am not sure how to init a `twitchio.PartialUser` object. 

I am a bit confused right now and not sure if I am on the right direction. Is it possible execute 'Twitch Chat Commands' via twitchio? It seems `twitchio.PartialUser` has all the functions that can moderate chats, but how to access it from the bot. Any insight is appreciated. Thank you.

**Comment by IAmTomahawkx** [comment, marked as answer]
Hi there, twitch has removed the ability for bots to use slash commands, unfortunately.
Here's a list of things you can use instead:
`/w` -> PartialUser.send_whisper
`/mod` -> PartialUser.add_channel_moderator / PartialUser.remove_channel_moderator
`/vip` -> PartialUser.add_channel_vip / PartialUser.remove_channel_vip
`/shoutout` -> PartialUser.shoutout
`/raid` -> PartialUser.start_raid (also: PartialUser.cancel_raid)
`/ban` -> PartialUser.ban_user
`/unban` -> PartialUser.unban_user
`/shield` ->  PartialUser.update_shield_mode_status
`/timeout` -> PartialUser.timeout_user
`/announce` -> PartialUser.chat_announcement
`/clear` / `/delete` -> PartialUser.delete_chat_messages
`/uniquechat`, `/followers`, `/emoteonly` -> PartialUser.update_chat_settings

`/host` will not have a replacement as it's being removed altogether
`/me` will continue to work
`/disconnect` will continue to work

**Reply by jetfuel** [reply]
Thank you for the reply. 

It's too bad Twitch removed the abilities to send Chat Commands. But I am glad we still have `PartialUser` to work with.

I also found a way to get a `PartialUser ` instance within the `Bot event_message` function. 
```
user = self.create_user(user_id, user_name)
await user.chat_announcement(token, user_id, 'Hello World')
```
